src/kernel/src/ucred/mod.rs

```diff
@@ -1,25 +1,41 @@
-pub use self::auth::*;
-pub use self::id::*;
-pub use self::privilege::*;
+use crate::errno::EAFNOSUPPORT;
+use crate::errno::ESRCH;
 use crate::errno::{Errno, EPERM};
+use crate::net::AddressFamily;
+use crate::ucred::prison::PrisonAllow;
+use crate::ucred::prison::PrisonFlags;
 use std::num::NonZeroI32;
+use std::sync::Arc;
 use thiserror::Error;
 
+pub use self::auth::*;
+pub use self::id::*;
+pub use self::prison::*;
+pub use self::privilege::*;
+
 mod auth;
 mod id;
+mod prison;
 mod privilege;
 
 /// An implementation of `ucred` structure.
 #[derive(Debug, Clone)]
 pub struct Ucred {
-    effective_uid: Uid, // cr_uid
-    real_uid: Uid,      // cr_ruid
-    groups: Vec<Gid>,   // cr_groups + cr_ngroups
+    effective_uid: Uid,  // cr_uid
+    real_uid: Uid,       // cr_ruid
+    groups: Vec<Gid>,    // cr_groups + cr_ngroups
+    prison: Arc<Prison>, // cr_prison
```

Comment:
Can we try something like `Cow<'static, Prison>` then implement `ToOwned` on `Prison`? If it works we can get rid of `lazy_static`.

Reply:
I didn't think of that at all.

Reply:
I tried something. Don't know if it's correct